lib/shared/src/models/sync.ts

```diff
@@ -196,54 +197,89 @@ export function syncModels({
+                                                if (
+                                                    isDeepCodyEnabled &&
+                                                    sonnetModel &&
+                                                    // Ensure the deep-cody model is only added once.
+                                                    !data.primaryModels.some(m =>
+                                                        m.id.includes('deep-cody')
+                                                    )
+                                                ) {
+                                                    const DEEPCODY_MODEL =
+                                                        getExperimentalClientModelByFeatureFlag(
+                                                            FeatureFlag.DeepCody
+                                                        )!
+                                                    data.primaryModels.push(
+                                                        ...maybeAdjustContextWindows([
+                                                            DEEPCODY_MODEL,
+                                                        ]).map(createModelFromServerModel)
+                                                    )
+                                                }
+                                                const isFreeTier =
+                                                    userSubscription !== null &&
+                                                    typeof userSubscription === 'object' &&
+                                                    'userCanUpgrade' in userSubscription &&
+                                                    userSubscription.userCanUpgrade
 
-                                            return Observable.of(data)
-                                        })
+                                                // set the default model to Haiku for free users
+                                                if (isDotComUser && isFreeTier && defaultToHaiku) {
+                                                    const haikuModel = data.primaryModels.find(m =>
+                                                        m.id.includes('claude-3-5-haiku')
+                                                    )
+                                                    if (haikuModel) {
+                                                        if (!data.preferences) {
+                                                            data.preferences = {
+                                                                defaults: {},
+                                                            }
+                                                        }
+                                                        data.preferences.defaults.chat = haikuModel.id
+                                                    }
```

Comment:
```suggestion
                                                    if (haikuModel) {
                                                        data.preferences!.defaults.chat = haikuModel.id
                                                    }
```

nit: we can simplify this because we use serverModelsConfig in DotCom, which includes the data.preferences fields.

Reply:
Nice catch!